_posts/yyyy-mm-dd-ansible-czy-da-sie-uporzadkowac-chaos.md

```diff
@@ -0,0 +1,301 @@
+    	find $BINS/tomcat/ -maxdepth 1 -mindepth 1 -not -name "webapps" | xargs rm -rf
+    
+   	log "Usuwanie aplikacji bez knowledgebase"
+    	find $BINS/tomcat/webapps/ -maxdepth 1 -mindepth 1 -not -name "knowledgebase*" | 	xargs rm -rf
+   
+   	log "Czyszczenie tomcata 9 na JVM 11.\nUsuwanie plików i katalogów bez webapps"
+    	find $BINS/tomcat-jvm11/ -maxdepth 1 -mindepth 1 -not -name "webapps" | xargs rm -rf
+    
+   	log "Usuwanie aplikacji bez knowledgebase"
+    	find $BINS/tomcat-jvm11/webapps/ -maxdepth 1 -mindepth 1 -not -name "knowledgebase*" | 	xargs rm -rf
+   
+    	log "Usuwanie: $BINS/eximee/bundles/\*"
+    	rm -rf $BINS/eximee/bundles/* || fail "Blad przy czyszczeniu poprzedniej instalacji"
+    
+   	log "Usuwanie: $BINS/eximee/bundles-jvm11/\*"
+    	rm -rf $BINS/eximee/bundles-jvm11/* || fail "Blad przy czyszczeniu poprzednieji instalacji"
+    
+   	if [ -d $BINS/servicemix/data/cache/bundle212 ]; then
+     		log "Usuwanie: $BINS/servicemix/data/cache/bundle212"
+     		rm -rf $BINS/servicemix/data/cache/bundle212 || fail "Blad przy czyszczeniu 				poprzedniej instalacji"
+    	fi
+    
+   	log $PHASE_SEPARATOR
+    
+   	log "Czyszczenie zalezne od platformy"
+    	cleanSystemBased || fail "Blad czyszczenia czesci zaleznej od platformy"
+    	log $PHASE_SEPARATOR
+   }
+```
+Niby nic nadzwyczajnego, a jednak na samo logowanie postępu i usunięcie plików, które i tak zostaną ponownie wgrane na środowisko wraz z nową paczką, przeznaczono jakieś 30 linii kodu. Choć logowanie niekoniecznie było niezbędne, to jednak pozwalało śledzić kolejne etapy procesu instalacji, dając większą wiedzę osobom analizującym ewentualne przyczyny nieudanej instalacji.
+
+W takim razie co miał zrobić administrator systemu, który chciałby skrócić proces i zainstalować tylko te komponenty, które uległy zmianie? No cóż, musiał on wykazać się nie lada cierpliwością i pewnością siebie, modyfikując dołączony do paczki skrypt instalacyjny lub po prostu pogodzić się z faktem, że nie da się tego zrobić „na skróty”. 
+
+Jakby tego było mało, chcąc zainstalować kompletną platformę na „świeżym” środowisku konieczna była praca z kilkoma archiwami zip, które zawierały poszczególne składowe systemu. Osobno bowiem dostarczano strukturę bazy danych, frontend, backend czy aplikacje do zarządzania systemem plików. W skrócie, aby administrator zainstalował platformę Eximee musiał co najmniej 4 razy powtórzyć podobny proces dla każdej części platformy. 
+
+# Era Ansible
+Już od wczesnych lat studiów programistom wpaja się, aby dążyć do utrzymania eleganckiej i czytelnej struktury swoich aplikacji (`KISS – Keep it simple, stupid`). Doskonale w ten trend wpasowuje się **Ansible**. To kupione przez firmę **Red Hat** opensource’owe oprogramowanie m.in. do automatyzacji procesu wdrażania aplikacji i zarządzania konfiguracją. W prosty sposób, za pomocą języka **YAML** pozwala opisać wzajemne relacje między systemami. 
```

Comment:
W prosty sposób, za pomocą języka **YAML**, pozwala opisać wzajemne relacje między systemami. (dodatkowy przecinek, bo "za pomocą języka YAML" to wtrącenie)
**albo i chyba nawet lepiej:**
Za pomocą języka **YAML** w prosty sposób pozwala opisać wzajemne relacje między systemami.

Reply:
poprawione